DataFormats/GeometrySurface/interface/ReferenceCounted.h

```diff
@@ -34,28 +32,24 @@ class BasicReferenceCounted
    public:
       BasicReferenceCounted() : referenceCount_(0) {}
       BasicReferenceCounted( const BasicReferenceCounted& /* iRHS */) : referenceCount_(0) {}
+      BasicReferenceCounted(BasicReferenceCounted&&) = default;
+      BasicReferenceCounted& operator=(BasicReferenceCounted&& ) = default;
 
-      const BasicReferenceCounted& operator=( const BasicReferenceCounted& ) {
+      BasicReferenceCounted& operator=( const BasicReferenceCounted& ) {
```

Comment:
The syntax of a assign operator is this one.
All this exists because reference counted objects may also be used as normal objects.
Otherwise we can simply delete copy and assignment if used only in refcoutingpointers...